src/utils/helpers.ts

```diff
@@ -29,15 +27,15 @@ export const getTimeGroup = (timestamp: number): string => {
 
 /**
  * Groups an array of conversations by time periods and sorts them by timestamp
- * @param conversations - A record of conversation histories keyed by their id
- * @returns An object with time period keys and arrays of conversations as values
+ * @param conversations - A record of conversation histories keyed by their id, can be either ConversationHistories or SearchableChatHistories
+ * @returns An object with time period keys (e.g., "Today", "Yesterday") and arrays of conversations as values
  */
-export const groupConversationsByTime = (
-  conversations: Record<string, ConversationHistory>,
-): GroupedConversations => {
+export const groupConversationsByTime = <T extends { lastSaved: number }>(
+  conversations: Record<string, T>,
+): Record<string, T[]> => {
```

Comment:
As long as `T` includes `lastSaved`, then allow that type to flow through the function